x-pack/metricbeat/module/aws/ec2/ec2_integration_test.go

```diff
@@ -0,0 +1,119 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+// +build integration
+
+package ec2
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/elastic/beats/metricbeat/mb"
+	mbtest "github.com/elastic/beats/metricbeat/mb/testing"
+)
+
+func TestFetch(t *testing.T) {
+	accessKeyID, okAccessKeyID := os.LookupEnv("AWS_ACCESS_KEY_ID")
+	secretAccessKey, okSecretAccessKey := os.LookupEnv("AWS_SECRET_ACCESS_KEY")
+	sessionToken, okSessionToken := os.LookupEnv("AWS_SESSION_TOKEN")
+	defaultRegion, _ := os.LookupEnv("AWS_REGION")
+
+	if !okAccessKeyID || accessKeyID == "" {
+		t.Skip("Skipping TestFetch; $AWS_ACCESS_KEY_ID not set or set to empty")
+	} else if !okSecretAccessKey || secretAccessKey == "" {
+		t.Skip("Skipping TestFetch; $AWS_SECRET_ACCESS_KEY not set or set to empty")
+	} else {
+		tempCreds := map[string]interface{}{
+			"module":                "aws",
+			"period":                "300s",
+			"metricsets":            []string{"ec2"},
+			"aws_access_key_id":     accessKeyID,
+			"aws_secret_access_key": secretAccessKey,
+			"aws_default_region":    defaultRegion,
+		}
+
+		if okSessionToken && sessionToken != "" {
+			tempCreds["aws_session_token"] = sessionToken
+		}
+
+		awsMetricSet := mbtest.NewReportingMetricSetV2(t, tempCreds)
+		events, errs := mbtest.ReportingFetchV2(awsMetricSet)
+		if errs != nil {
+			t.Skip("Skipping TestFetch: failed to make api calls. Please check $AWS_ACCESS_KEY_ID, " +
+				"$AWS_SECRET_ACCESS_KEY and $AWS_SESSION_TOKEN in config.yml")
+		}
+
+		assert.Empty(t, errs)
+		if !assert.NotEmpty(t, events) {
+			t.FailNow()
+		}
+		t.Logf("Module: %s Metricset: %s", awsMetricSet.Module().Name(), awsMetricSet.Name())
+
+		for _, event := range events {
+			fmt.Println("event = ", event)
+			// RootField
+			checkEventField("service.name", "string", event, t)
```

Comment:
Alternatively what I think should also work here is creating a `common.MapStr` with all the fields inside and then compare it. Something like:

```
result := common.MapStry{
  "cloud": common.MapStr{
    "instance": common.MapStr{
      "id": "foo"
    }
  }
}
```

The type would then also be directly compared if it is correct.

Reply:
@ruflin Your suggestion will work for TestMockFetch in ec2_test.go but probably won't work here in integration test though. In this integration test, it will go to an actual aws account to query for ec2 instance information and cloudwatch metrics with real data which the value will vary. Am I misunderstanding your suggestion?

Reply:
No, that sounds correct.